### PR TITLE
Toniof xxx improve preprocessing

### PR DESCRIPTION
**Pull request by TonioF**
Mimicing a real life case where
- a time variable was present but no time dim was added to the variables
- time bounds had a unique name (date_bnds)
- time bounds were a variable, not a coordinate

### CODE REVIEW

File: tests/test_preprocessor.py [hunk 1]
```diff
@@ -32,6 +32,20 @@
 
 class DatasetPreProcessorTest(unittest.TestCase):
 
+    def test_time_must_be_expanded_time_bounds_as_var(self):
```
Comment (forman) [inline tests/test_preprocessor.py:35]: Please fix/change test name, this is not clear.

File: tests/test_preprocessor.py [hunk 2]
```diff
@@ -32,6 +32,20 @@
...
+                              add_time_as_dim_to_vars=False,
+                              time_bounds_name='date_bnds',
+                              add_time_bounds_as_var=True)
+        pre_processor = DatasetPreProcessor(input_concat_dim='time')
+        ds = pre_processor.preprocess_dataset(ds)
+        self.assertIn('date_bnds', ds)
+        self.assertIn('date_bnds', ds.coords)
+        self.assertEqual(('time', 'bnds'), ds.date_bnds.dims)
+        self.assertIn('r_ui16', ds)
+        self.assertEqual(('time', 'lat', 'lon'), ds.r_ui16.dims)
```
Comment (forman) [inline tests/test_preprocessor.py:47]: You should still assert that the "time" coordinate

1. is in ds, 
2. a 1-D array, 
3. has "bounds" attribute and its value is "date_bnds".

**Review by forman — CHANGES_REQUESTED** [review]
Good! But please

* record change in `CHANGES.md`
* extend tests as indicated.

**Review by forman — APPROVED** [review]

**Comment by forman** [comment]
Thanks @TonioF

**Review by pont-us — APPROVED** [review]

**Pull request merged**